src/pymovements/dataset/dataset.py

```diff
@@ -231,6 +231,40 @@ def load_precomputed_reading_measures(self) -> None:
             self.paths,
         )
 
+    def _split_gaze_data(
```

Comment:
can you move most of the logic into a `GazeDataFrame` method like this?

```python
def split(self, by: list[str] | str) -> list[GazeDataFrame]:
```

You can reuse it in the `Dataset` method

Reply:
not so easy, especially for datasets we want to split the file s.t. fileinfo is also duplicated -- which is not present in GazeDataFrame.